internal/subscriber/retry/delayconsumer_test.go

```diff
@@ -81,29 +116,177 @@ func TestDelayConsumer_Run(t *testing.T) {
 	dc.Run(ctx)
 }
 
-func getDummyBrokerMessage() messagebroker.ReceivedMessage {
-	tNow := time.Now()
-	tPast := tNow.Add(time.Second * 100 * -1)
+func TestDelayConsumer_Run_Consume(t *testing.T) {
+	// message with delay within duration of test-case
+	msg1 := getDummyBrokerMessage("msg-1", uuid.New().String(), time.Now().Add(time.Nanosecond*1000), 5)
+	// message with delay outside duration of test-case
+	msg2 := getDummyBrokerMessage("msg-2", uuid.New().String(), time.Now().Add(time.Second*2), 5)
+
+	type Test struct {
+		name                 string
+		messages             *messagebroker.ReceivedMessage
+		expectedMessage      *messagebroker.ReceivedMessage
+		expectedMessageCount int
+	}
+
+	tests := []Test{
+		{
+			name:                 "DelayConsumer with message delivery time within test duration",
+			messages:             &msg1,
+			expectedMessage:      nil,
+			expectedMessageCount: 0,
+		},
+		{
+			name:                 "DelayConsumer with message delivery time above test duration",
+			messages:             &msg2,
+			expectedMessage:      &msg2,
+			expectedMessageCount: 1,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// since new context is needed to for each time so set up is called each time
+			ctx, ctrl, subCtx, cancel, brokerStore, handler, consumer, cache := setupDC(t)
+			subs := getMockSubscriptionModel()
+			subscriberID := "subscriber-id"
+			errCh := make(chan error)
+			dc, err := NewDelayConsumer(subCtx, subscriberID, "t1", subs, brokerStore, handler, cache, errCh)
+			assert.NotNil(t, dc)
+			assert.Nil(t, err)
+			assert.NotNil(t, dc.LogFields())
+
+			producer := getMockProducer(ctx, ctrl)
+			brokerStore.EXPECT().GetProducer(gomock.Any(), gomock.Any()).Return(producer, nil).AnyTimes()
+
+			go dc.Run(ctx)
+
+			messages := make([]messagebroker.ReceivedMessage, 0)
+			messages = append(messages, *test.messages)
+			count := 0
+			consumer.EXPECT().ReceiveMessages(gomock.AssignableToTypeOf(subCtx), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: messages}, nil
+					} else {
+						return &messagebroker.GetMessagesFromTopicResponse{}, nil
+					}
+				}).AnyTimes()
+
+			select {
+			case err := <-errCh:
+				assert.Fail(t, err.Error())
+			case <-time.NewTicker(time.Millisecond * 500).C:
+				assert.Equal(t, len(dc.cachedMsgs), test.expectedMessageCount)
+				if len(dc.cachedMsgs) != 0 {
+					assert.True(t, reflect.DeepEqual(dc.cachedMsgs[0], *test.expectedMessage))
```

Comment:
rather than comparing just the first message from cachedMsgs, compare the entire slice

Reply:
changed to compare entire slice